source/common/upstream/cluster_manager_impl.cc

```diff
@@ -170,21 +170,25 @@ ClusterManagerImpl::ClusterManagerImpl(const envoy::config::bootstrap::v2::Boots
                                        const LocalInfo::LocalInfo& local_info,
                                        AccessLog::AccessLogManager& log_manager,
                                        Event::Dispatcher& main_thread_dispatcher,
-                                       Server::Admin& admin)
+                                       Server::Admin& admin, SystemTimeSource& system_time_source,
+                                       MonotonicTimeSource& monotonic_time_source)
     : factory_(factory), runtime_(runtime), stats_(stats), tls_(tls.allocateSlot()),
       random_(random), bind_config_(bootstrap.cluster_manager().upstream_bind_config()),
       local_info_(local_info), cm_stats_(generateStats(stats)),
       init_helper_([this](Cluster& cluster) { onClusterInit(cluster); }),
       config_tracker_entry_(
-          admin.getConfigTracker().add("clusters", [this] { return dumpClusterConfigs(); })) {
+          admin.getConfigTracker().add("clusters", [this] { return dumpClusterConfigs(); })),
+      system_time_source_(system_time_source), monotonic_time_source_(monotonic_time_source) {
```

Comment:
`monotonic_time_source_` seems to be unused?

Reply:
Yep, this is what a/tsan are failing on. Will fix.